analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationDispatcher.cs

```diff
@@ -67,10 +67,15 @@ public static IEnumerable<SymbolicContext> Process(SymbolicContext context)
     {
         if (Simple.TryGetValue(context.Operation.Instance.Kind, out var simple))            // Operations that return single state
         {
-            context = context.WithState(simple.Process(context));
+            return new[] { context.WithState(simple.Process(context)) };
```

Comment:
Unrelated improvement. If an operation is registered in the 1st list, there's no reason look in the 2nd one too

Reply:
Do we have some kind of test or check, so that we don't end up with one operation kind in both lists?

Reply:
No, we don't. It should be easy to catch, as the change goes for "simple" first, so whatever complex logic we'll have should just not work. And should be easy to catch